dace/frontend/fortran/fortran_parser.py

```diff
@@ -1,58 +1,314 @@
-# Copyright 2023 ETH Zurich and the DaCe authors. All rights reserved.
+# Copyright 2019-2025 ETH Zurich and the DaCe authors. All rights reserved.
 
-from venv import create
+import copy
+import os
+import sys
 import warnings
+from copy import deepcopy as dpcp
+from pathlib import Path
+from typing import List, Optional, Set, Dict, Tuple, Union, Iterable
 
-from dace.data import Scalar
+import networkx as nx
+from fparser.api import get_reader
+from fparser.two.C99Preprocessor import CPP_CLASS_NAMES
+from fparser.two.Fortran2003 import Program, Module_Stmt, Include_Stmt, Subroutine_Stmt, Function_Stmt, Interface_Stmt, \
+    Execution_Part, Component_Decl
+from fparser.two.parser import ParserFactory
+from fparser.two.utils import Base, walk, FortranSyntaxError
 
 import dace.frontend.fortran.ast_components as ast_components
-import dace.frontend.fortran.ast_transforms as ast_transforms
-import dace.frontend.fortran.ast_utils as ast_utils
 import dace.frontend.fortran.ast_internal_classes as ast_internal_classes
-from typing import List, Optional, Tuple, Set
-from dace import dtypes
+import dace.frontend.fortran.ast_transforms as ast_transforms
 from dace import Language as lang
+from dace import SDFG, InterstateEdge, Memlet, pointer, SDFGState
 from dace import data as dat
-from dace import SDFG, InterstateEdge, Memlet, pointer, nodes
+from dace import dtypes
 from dace import symbolic as sym
-from dace.sdfg.state import ControlFlowRegion, LoopRegion
-from copy import deepcopy as dpcp
-
+from dace.data import Scalar, Structure
+from dace.frontend.fortran import ast_utils
+from dace.frontend.fortran.ast_desugaring import ENTRY_POINT_OBJECT_CLASSES, NAMED_STMTS_OF_INTEREST_CLASSES, SPEC, \
```

Comment:
at this point, just import desugaring and call internal functions. Look at our Contributing Guidelines to see that we do not import classes and functions unless they are a very specific bunch (Memlet etc.).